e2etests/run_athens.go

```diff
@@ -0,0 +1,56 @@
+// +build e2etests
+
+package e2etests
+
+import (
+	"fmt"
+	"net/http"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"time"
+)
+
+func buildAthens(goBin string, destPath string, env []string) (string, error) {
+	target := path.Join(destPath, "athens-proxy")
+	binFolder, err := filepath.Abs("../cmd/proxy")
+	if err != nil {
+		return "", fmt.Errorf("Failed to get athens source path %v", err)
+	}
+
+	cmd := exec.Command(goBin, "build", "-o", target, binFolder)
+	cmd.Env = env
+	err = cmd.Run()
+	return target, err
+}
+
+func stopAthens() error {
```

Comment:
I think a better way to "stopAthens" is to have the runAthensAndWait take a context that you can cancel whenever you want to stop athens and just pass that context into https://golang.org/pkg/os/exec/#CommandContext

which should render this whole function unnecessary.

Reply:
Did not know about it, thanks!

Reply:
Well, yes and no, as I am using stopAthens also to remove any dangling instance before running the real tests. So it's ok for the teardown phase, but I'll keep this around as well.